src/main/java/com/iri/dao3/dao/jdbc/CarDaoImpl.java

```diff
@@ -0,0 +1,228 @@
+            preparedStatement.setLong(2, car.getManufacturer().getId());
+            preparedStatement.setLong(3, car.getId());
+            preparedStatement.executeUpdate();
+            removeDriversFromCar(car);
+            for (Driver driver : car.getDrivers()) {
+                addDriverToCar(driver, car);
+            }
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't refresh car in method update", e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteQuery = "UPDATE cars SET deleted = true WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(deleteQuery)) {
+            preparedStatement.setLong(1, id);
+            return preparedStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't mark car in method delete", e);
+        }
+    }
+
+    @Override
+    public void addDriverToCar(Driver driver, Car car) {
+        String insertQuery = "INSERT INTO cars_drivers (driver_id, car_id) "
+                + "VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(insertQuery)) {
+            preparedStatement.setLong(1, driver.getId());
+            preparedStatement.setLong(2, car.getId());
+            preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't insert in method addDriverToCar", e);
+        }
+    }
+
+    @Override
+    public void removeDriverFromCar(Driver driver, Car car) {
+        String deleteQuery = "DELETE FROM `cars_drivers` WHERE car_id = ? AND driver_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(deleteQuery)) {
+            preparedStatement.setLong(1, car.getId());
+            preparedStatement.setLong(2, driver.getId());
+            preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete in method removeDriverFromCar", e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String selectQuery = "SELECT  c.id as id, c.model as model, "
+                + "m.id as manufacturer_id, m.name as manufacturer_name, "
+                + "m.country as manufacturer_country "
+                + "FROM cars_drivers cd "
+                + "INNER JOIN cars c ON cd.car_id = c.id "
+                + "INNER JOIN drivers d ON cd.driver_id = d.id "
+                + "INNER JOIN manufacturer m ON c.manufacturer_id = m.id "
+                + "WHERE c.deleted != TRUE AND d.deleted != TRUE AND m.deleted != TRUE";
```

Comment:
```suggestion
                + "WHERE c.deleted = FALSE AND d.deleted = FALSE AND m.deleted = FALSE";
```
and what about driver id?

Reply:
done

Reply:
done